### PR TITLE
Feat/aut 3623/hide features in solar design

### PR DESCRIPTION
**Pull request by shaveko**
related to 
 - https://oat-sa.atlassian.net/browse/AUT-3623
 - https://oat-sa.atlassian.net/browse/AUT-3625

⚠️ Update extension version to the release ones before merging

###Description

This PR addresses requirement `2.a` from the ticket to remove Test-Taker Panel from Deliveries page if solar design is activated.

### How to test

To test locally the dependent FE/BE branches should be specified in `composer.json`
 - [feat/AUT-3625/content-creation-mode-switch](https://github.com/oat-sa/generis/tree/feat/AUT-3625/content-creation-mode-switch)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/tao-core/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/extension-tao-deliver-connect/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/extension-tao-testqti/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [feat/AUT-3623/hide-features-in-solar-design](https://github.com/oat-sa/extension-tao-delivery-rdf/tree/feat/AUT-3623/hide-features-in-solar-design)
 - [feat/AUT-3623/hide-features-in-solar-design](https://github.com/oat-sa/extension-tao-itemqti/tree/feat/AUT-3623/hide-features-in-solar-design)

Deployed to kitchen env, so functionally can be validated there: http://test-rfe-1147-facelift.playground.kitchen.it.taocloud.org:49381/

### CODE REVIEW

**Review by gabrielfs7 — APPROVED** [review]
Great job! Approving the BE part here. Thanks!

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Pull request's target is not `master`

File: views/js/controller/DeliveryMgmt/editDelivery.js [hunk 1]
```diff
@@ -20,13 +20,56 @@
...
+            }
+            if(!features.isVisible(`${featuresPath}proctoringSettings`)) {
+                hidePropertyBlockByName('http_2_www_0_tao_0_lu_1_Ontologies_1_TAODelivery_0_rdf_3_ProctorAccessible_0');
+            }
+            if(!features.isVisible(`${featuresPath}publicationId`)) {
+                hidePropertyBlockByName('http_2_www_0_tao_0_lu_1_Ontologies_1_taoDeliverConnect_0_rdf_3_PublicationId');
+            }
+            if(!features.isVisible(`${featuresPath}assessmentProjectId`)) {
+                hidePropertyBlockByName('http_2_www_0_tao_0_lu_1_Ontologies_1_TAODelivery_0_rdf_3_AssessmentProjectId');
+            }
```
Comment (jsconan) [inline views/js/controller/DeliveryMgmt/editDelivery.js:71]: **quibble**: This code looks boilerplate and cumbersome...

**suggestion:** As the code is highly repetitive, a better approach is to create an array of couple id/uri, and pass it to a single helper that will do the recurrent work:
```javascript
features.forEach(feature => {
    if(!features.isVisible(`${featuresPath}${feature.id}`)) {
        hidePropertyBlockByName(feature.uri);
    }
})
```

**Review by jsconan — APPROVED** [review]
It works, but the code could be simplified for easier maintenance and more readability.

File: views/js/controller/DeliveryMgmt/editDelivery.js [hunk 2]
```diff
@@ -20,13 +20,56 @@
...
+    }
+
     return {
         start(){
+
+            const featuresPath = 'taoDeliveryRdf/deliveryMgmt/';
+            if(!features.isVisible(`${featuresPath}resourceIdentifier`)) {
+                hidePropertyBlockByName('id');
+            }
+            if(!features.isVisible(`${featuresPath}maxExecutions`)) {
```
Comment (gabrielfs7) [inline views/js/controller/DeliveryMgmt/editDelivery.js:45]: @shaveko this form is built in the BE side, I am not sure why we are hiding the fields in the FE. 

We should probably have done this in the BE https://github.com/oat-sa/extension-tao-delivery-rdf/blob/develop/model/Delivery/Presentation/Web/Form/DeliveryFormFactory.php#L53

I am also concerned on the side effects of removing it in the FE side, cause we are still considering those fields in the BE and I am not sure the integrity of the data will be consistent after submitting this form.

File: views/js/controller/DeliveryMgmt/editDelivery.js (hunk 1 above)
Comment (gabrielfs7) [reply views/js/controller/DeliveryMgmt/editDelivery.js:71]: @jsconan I am not sure it is a good idea to do this in the FE side https://github.com/oat-sa/extension-tao-delivery-rdf/pull/484/files#r1647140712

File: views/js/controller/DeliveryMgmt/editDelivery.js [hunk 3]
```diff
@@ -20,13 +20,56 @@
...
     'css!taoDeliveryRdfCss/delivery-rdf.css'
-], function ($, urlUtil) {
+], function ($, urlUtil, features) {
     'use strict';
 
+    /**
+     * Finds property block and hides it adding .hidden class
+     * @param {string} propName 
+     */
+    function hidePropertyBlockByName(propName) {
```
Comment (gabrielfs7) [inline views/js/controller/DeliveryMgmt/editDelivery.js:33]: @Nevermind23 @shaveko we are also missing here the `http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryTestRunnerFeatures`

Comment (gabrielfs7) [reply views/js/controller/DeliveryMgmt/editDelivery.js:33]: This is part of this very extension https://github.com/oat-sa/extension-tao-delivery-rdf/blob/c225cee92f19792d41e158a1c35897777069cab3/install/ontology/taodelivery.rdf#L74

**Pull request merged**